contractcourt/channel_arbitrator.go

```diff
@@ -802,7 +837,19 @@ func (c *ChannelArbitrator) stateStep(triggerHeight uint32,
 		// If the resolution is empty, then we're done here. We don't
 		// need to launch any resolvers, and can go straight to our
 		// final state.
-		if contractResolutions.IsEmpty() {
+		//
+		// TODO(roasbeef): not needed since we must have a commitment
+		// output in order to make an outgoing HTLC?
+		noCommitHtlcs := true
+		if confCommitSet != nil {
+			localHTLCs := len(confCommitSet.LocalCommit.Htlcs) > 0
+			remoteHTLCs := len(confCommitSet.RemoteCommit.Htlcs) > 0
+			danglingHtlcs := (confCommitSet.RemotePendingCommit != nil &&
+				len(confCommitSet.RemotePendingCommit.Htlcs) > 0)
+
+			noCommitHtlcs = !localHTLCs && !remoteHTLCs && !danglingHtlcs
```

Comment:
adding a helper method to compute this would be useful, e.g. `confCommitSet.IsEmpty()`

Reply:
`CommitSet.IsEmpty` is defined, but not called here?

Reply:
Ah messed up in the rebase somehow, should be fixed in the latest diff.